source/elements/oneDPL/source/parallel_api/parallel_range_api.rst

```diff
@@ -324,5 +440,79 @@ Mutating Operations
 
   }
 
+In-place Mutating Operations
+++++++++++++++++++++++++++++
+
+.. code:: cpp
+
+  // Defined in <oneapi/dpl/algorithm>
+
+  namespace oneapi::dpl::ranges {
+
+    // fill
+    template <typename ExecutionPolicy, std::ranges::random_access_range R,
+              typename T = std::ranges::range_value_t<R>>
+      requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<ExecutionPolicy>> &&
+               std::ranges::sized_range<R> &&
+               std::indirectly_writable<std::ranges::iterator_t<R>, const T&>
+      std::ranges::borrowed_iterator_t<R>
+        fill (ExecutionPolicy&& pol, R&& r, const T& value);
+  
+    // generate
+    template <typename ExecutionPolicy, std::ranges::random_access_range R, std::copy_constructible F>
+      requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<ExecutionPolicy>> &&
+               std::ranges::sized_range<R> && std::invocable<F&> &&
+               std::indirectly_writable<std::ranges::iterator_t<R>, std::invoke_result_t<F&>>
+      std::ranges::borrowed_iterator_t<R>
+        generate (ExecutionPolicy&& pol, R&& r, F gen);
+
+    // replace
+    template <typename ExecutionPolicy, std::ranges::random_access_range R,
+              typename Proj = std::identity,
+              typename T1 = std::projected_value_t<std::ranges::iterator_t<R>, Proj>, typename T2 = T1>
```

Comment:
I think a default type written as `std::projected_value_t` should be omitted(here and in the other places), because it available since C++26. 
Also that difference should be added to the differences list above.

Reply:
Cool. The problem is, we already use it for `count`, `find`, and `search_n`.

Reply:
For example,
https://github.com/uxlfoundation/oneDPL/pull/2100/files#diff-c7a5edf4a7b057df5179c8f71fb4701c66ea627fb1aaf8965e4c2e81f0c43a94R815

Reply:
The definition of `std::projected_value_t` only uses APIs available in C++20, so I think the proper resolution is to "define" a custom `projected_value_t` in the spec (and in the implementation as well).

Reply:
https://github.com/uxlfoundation/oneAPI-spec/pull/619 is the PR to fix the issue.